scipy/optimize/tests/test_linprog.py

```diff
@@ -1109,7 +1113,21 @@ class TestLinprogIPDense(BaseTestLinprogIP):
 class TestLinprogIPSparsePresolve(BaseTestLinprogIP):
     options = {"sparse": True, "_sparse_presolve": True}
 
+    @pytest.mark.skip(reason='_sparse_presolve=True incompatible '
+                      'with presolve=False')
```

Comment:
I don't quite understand this bit. Why add a test for things that are incompatible - this will always remain a skip then right?

Reply:
Yeah, this is actually the reason I asked you in particular to take a look. I wasn't sure whether this was an acceptable use of skip. 

Given the existing class structure, it seemed to be the simplest way to apply this test to simplex, interior-point dense, and interior-point sparse without applying the test to interior point sparse with sparse presolve (because presolve is False, so sparse presolve doesn't make sense). 

So yes, it would always be skipped. 
Should I just remove this test from all the classes and perform it for the three cases?

Reply:
Ah okay. This isn't very clear, I suggest to keep the test here but implement it as:

    def test_enzo_example_c_with_infeasibility(self):
         pytest.skip('_sparse_presolve=True incompatible with presolve=False')